pkgs/servers/mail/dovecot/default.nix

```
@@ -22,7 +22,7 @@ stdenv.mkDerivation rec {
 
   src = fetchurl {
     url = "https://dovecot.org/releases/2.3/${pname}-${version}.tar.gz";
```

```suggestion
    url = "https://dovecot.org/releases/${lib.versions.majorMinor version}/${pname}-${version}.tar.gz";
```
https://github.com/NixOS/nixpkgs/blob/102eb68ceecbbd32ab1906a53ef5a7269dc9794a/lib/versions.nix#L38-L47

---

I don't except them to release 2.4 anytime soon, but why not. Also soft-coded it in pigeonhole.